src/contracts/GPv2Settlement.sol

```diff
@@ -140,6 +142,69 @@ contract GPv2Settlement is GPv2Signing, ReentrancyGuard, StorageAccessible {
         emit Settlement(msg.sender);
     }
 
+    /// @dev Settle an order directly against Balancer V2 pools.
+    function swap(
+        IVault.SwapRequest[] calldata swaps,
+        IERC20[] calldata tokens,
+        GPv2Trade.Data calldata trade
+    ) external nonReentrant onlySolver {
+        RecoveredOrder memory recoveredOrder = allocateRecoveredOrder();
+        GPv2Order.Data memory order = recoveredOrder.data;
+        recoverOrderFromTrade(recoveredOrder, tokens, trade);
+
+        IVault.SwapKind kind =
+            order.kind == GPv2Order.SELL
+                ? IVault.SwapKind.GIVEN_IN
+                : IVault.SwapKind.GIVEN_OUT;
+
+        IVault.FundManagement memory funds;
+        funds.sender = recoveredOrder.owner;
+        funds.fromInternalBalance = order.useInternalSellTokenBalance;
+        funds.recipient = recoveredOrder.receiver;
+        funds.toInternalBalance = order.useInternalBuyTokenBalance;
+
+        int256[] memory limits = new int256[](tokens.length);
+        // NOTE: Array allocation initializes elements to 0, so we only need to
+        // set the limits we care about.
+        limits[trade.sellTokenIndex] = order.sellAmount.toInt256();
+        limits[trade.buyTokenIndex] = -order.buyAmount.toInt256();
+
+        GPv2Transfer.Data memory feeTransfer;
+        feeTransfer.account = recoveredOrder.owner;
+        feeTransfer.token = order.sellToken;
+        feeTransfer.amount = order.feeAmount;
+        feeTransfer.useInternalBalance = order.useInternalSellTokenBalance;
+
+        int256[] memory tokenDeltas =
+            vaultRelayer.batchSwapWithFee(
+                kind,
+                swaps,
+                tokens,
+                funds,
+                limits,
+                order.validTo,
+                feeTransfer
+            );
+
+        // NOTE: Check that the orders were completely filled and update their
+        // filled amounts to avoid replaying them.
+        require(filledAmount[recoveredOrder.uid] == 0, "GPv2: order filled");
```

Comment:
Is there a technical reason why we cannot support partially fillable orders using the fastpath?

Reply:
Simplicity, and, from what I understood, partially fillable orders aren't really intended to be used for swaps. But no, there is no technical reason.